frontend/src/employee-frontend/components/income-statements/IncomeStatementsPage.tsx

```diff
@@ -58,20 +63,67 @@ function IncomeStatementsList({
 export default React.memo(function IncomeStatementsPage() {
   const { i18n } = useTranslation()
 
-  const [result, setResult] = useState<
-    Result<IncomeStatementAwaitingHandler[]>
+  const [areas, setAreas] = useState<Result<CareArea[]>>(Loading.of())
+
+  const [incomeStatements, setIncomeStatements] = useState<
+    Result<Paged<IncomeStatementAwaitingHandler> & { currentPage: number }>
   >(Loading.of())
-  const loadData = useRestApi(getIncomeStatementsAwaitingHandler, setResult)
-  useEffect(() => loadData(), [loadData])
+  const [toggledAreas, setToggledAreas] = useState<string[]>([])
+
+  const loadAreas = useRestApi(getAreas, setAreas)
+  const loadData = useRestApi(
+    getIncomeStatementsAwaitingHandler,
+    setIncomeStatements
+  )
+
+  useEffect(() => {
+    loadAreas()
+    loadData()
+  }, [loadAreas, loadData])
+
+  useEffect(() => {
+    loadData(toggledAreas)
+  }, [loadData, toggledAreas])
+
+  const toggleArea = (area: string) => () => {
+    setToggledAreas((prev) => {
+      if (prev.includes(area)) return prev.filter((t) => t !== area)
+      else return [...prev, area]
+    })
+  }
```

Comment:
Just a personal preference but I would prefer ternary instead if-else and would omit the braces.


```ts
const toggleArea = (area: string) => () =>
  setToggledAreas((prev) =>
    prev.includes(area) ? prev.filter((t) => t !== area) : [...prev, area]
  )
```

Reply:
Good catch!